lib/services/local/assistant.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	msg, err := s.getConversation(ctx, username, request.Id)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Only update the title, leave the rest of the fields intact.
+	msg.Title = request.Title
+
+	payload, err := json.Marshal(msg)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	item := backend.Item{
+		Key:   backend.Key(assistantConversationPrefix, username, request.Id),
+		Value: payload,
+	}
+
+	if _, err = s.Update(ctx, item); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// GetAssistantConversations returns all conversations started by a user.
+func (s *IdentityService) GetAssistantConversations(ctx context.Context, username string, _ *proto.GetAssistantConversationsRequest) (*proto.GetAssistantConversationsResponse, error) {
+	startKey := backend.Key(assistantConversationPrefix, username)
+	result, err := s.GetRange(ctx, startKey, backend.RangeEnd(startKey), backend.NoLimit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	conversationsIDs := make([]*proto.ConversationInfo, 0, len(result.Items))
+	for _, item := range result.Items {
+		payload := &Conversation{}
+		if err := json.Unmarshal(item.Value, payload); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		conversationsIDs = append(conversationsIDs, &proto.ConversationInfo{
+			Id:          payload.ConversationID,
+			Title:       payload.Title,
+			CreatedTime: payload.CreatedTime,
+		})
+	}
+
+	sort.Slice(conversationsIDs, func(i, j int) bool {
+		return conversationsIDs[i].CreatedTime.Before(conversationsIDs[j].GetCreatedTime())
+	})
+
+	return &proto.GetAssistantConversationsResponse{
+		Conversations: conversationsIDs,
+	}, nil
+}
+
+// GetAssistantMessages returns all messages with given conversation ID.
+func (s *IdentityService) GetAssistantMessages(ctx context.Context, username, conversationId string) (*proto.GetAssistantMessagesResponse, error) {
+	if username == "" {
+		return nil, trace.BadParameter("missing username")
+	}
+
+	if conversationId == "" {
+		return nil, trace.BadParameter("missing conversation ID")
+	}
+
+	startKey := backend.Key(assistantMessagePrefix, username, conversationId)
+	result, err := s.GetRange(ctx, startKey, backend.RangeEnd(startKey), backend.NoLimit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	out := make([]*proto.AssistantMessage, len(result.Items))
+	for i, item := range result.Items {
+		var a proto.AssistantMessage
+		if err := json.Unmarshal(item.Value, &a); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		out[i] = &a
+	}
+
+	sort.Slice(out, func(i, j int) bool {
+		// Sort by created time.
+		return out[i].CreatedTime.Before(out[j].GetCreatedTime())
+	})
+
+	return &proto.GetAssistantMessagesResponse{
+		Messages: out,
+	}, nil
+}
+
+// CreateAssistantMessage adds the message to the backend.
+func (s *IdentityService) CreateAssistantMessage(ctx context.Context, username string, msg *proto.AssistantMessage) error {
```

Comment:
So here you have `CreateAssistantMessage` but the proto service method is called `InsertAssistantMessage` as I mentioned above. I would consider making the naming consistent and just using `Create...`.

Reply:
Renamed